### DISCUSSION TITLE
How to store Product "Variations"?

### DISCUSSION THREAD

**Discussion by oguruma1218** [supabase/supabase#14402] [Questions]
I have an app that has a Products table. Most Products are unique, however some of them have variations which make them something of a different product. For example, a pair of boots called "The Luxor" typically come in suede, however there's a special edition "The Luxor" boot that comes in an expensive Italian leather. 

What's the best way to store the original "Base" Product and the variant of that product? 

Do I create a separate table called Variants and then relate the Variants to their parent Product?

**Comment by GaryAustin1** [comment]
Although a Supabase user may have advice for you this question is not limited, to Supabase or even Postgres.
You probably should try a broader internet search on something like `relational database how to store product variants` and then when you have a structure that meets your needs, narrow down the Postgres code needed to generate it.

You'll get hits like this:
https://stackoverflow.com/questions/19144200/designing-a-sql-schema-for-a-combination-of-many-to-many-relationship-variation
